packages/mask/src/plugins/Predict/Azuro/PlaceBetDialog.tsx

```diff
@@ -0,0 +1,235 @@
+import { makeStyles } from '@masknet/theme'
+import { useI18N } from '../../../utils'
+import { isGreaterThan, rightShift, isZero, NetworkPluginID } from '@masknet/web3-shared-base'
+import { useAzuroConstants, TransactionStateType, ChainId } from '@masknet/web3-shared-evm'
+import { useChainId, useFungibleToken, useFungibleTokenBalance } from '@masknet/plugin-infra/web3'
+import { useState, useMemo, useEffect, useCallback } from 'react'
+import { AzuroGame, RATE_DECIMALS, USDT_DECIMALS } from '@azuro-protocol/sdk'
+import type { Odds } from './types'
+import { WalletConnectedBoundary } from '../../../web3/UI/WalletConnectedBoundary'
+import { EthereumERC20TokenApprovedBoundary } from '../../../web3/UI/EthereumERC20TokenApprovedBoundary'
+import ActionButton from '../../../extension/options-page/DashboardComponents/ActionButton'
+import { useBoolean } from 'react-use'
+import { marketRegistry, outcomeRegistry } from './helpers'
+import { usePlaceBetCallback, useMinRate, useActualRate } from './hooks'
+import { ConfirmModal } from '../../Tips/components/common/ConfirmModal'
+import { activatedSocialNetworkUI } from '../../../social-network'
+import { SuccessIcon } from '@masknet/icons'
+import { isTwitter } from '../../../social-network-adaptor/twitter.com/base'
+import { isFacebook } from '../../../social-network-adaptor/facebook.com/base'
+import { contractAddresses } from '../constants'
+
+const useStyles = makeStyles()((theme) => ({
+    root: {
+        fontFamily: 'Muli,Helvetica,-apple-system,system-ui,"sans-serif"',
+        width: '100%',
+        boxShadow: 'none',
+        border: `solid 1px ${theme.palette.divider}`,
+        padding: 0,
+    },
+    content: {
+        width: '100%',
+        height: 'var(--contentHeight)',
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'space-around',
+        padding: '0 !important',
+    },
+    walletStatusBox: { margin: theme.spacing(1, 1, 3, 1) },
+    container: { padding: theme.spacing(1) },
+    infosContainer: {
+        padding: theme.spacing(1.75, 0.75, 0.75, 0.75),
+        gap: theme.spacing(0.5),
+    },
+    infoContainer: {
+        minHeight: 30,
+    },
+    infoTitle: {
+        fontWeight: '300',
+    },
+    actionButton: {
+        flexDirection: 'column',
+        position: 'relative',
+        marginTop: theme.spacing(1.5),
+        lineHeight: '22px',
+        justifyContent: 'center',
+        alignItems: 'center',
+        padding: '13px 0',
+        fontSize: 18,
+    },
+}))
+
+export interface PlaceBetDialogProps {
+    open: boolean
+    onClose?: () => void
+    game: AzuroGame
+    condition: Odds
+}
+
+export function PlaceBetDialog(props: PlaceBetDialogProps) {
+    const { open, onClose, game, condition } = props
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const [amount, setAmount] = useState('')
+    const [slippage, setSlippage] = useState<number>(2)
```

Comment:
Generic argument should be omitted for primitive types.
```suggestion
    const [slippage, setSlippage] = useState(2)
```

Reply:
Done.